### PR TITLE
Don't disable resource leak detector if set by user

### PR DESCRIPTION
**Pull request by jvican**
Fixes https://github.com/eclipse-vertx/vert.x/issues/3616

### CODE REVIEW

File: src/main/java/io/vertx/core/impl/VertxImpl.java [hunk 1]
```diff
@@ -86,8 +86,12 @@
   private static final int NETTY_IO_RATIO = Integer.getInteger(NETTY_IO_RATIO_PROPERTY_NAME, 50);
 
   static {
-    // Netty resource leak detection has a performance overhead and we do not need it in Vert.x
-    ResourceLeakDetector.setLevel(ResourceLeakDetector.Level.DISABLED);
+    // Disable Netty's resource leak detection to reduce the performance overhead if not set by user
+    if (System.getProperty("io.netty.leakDetection.level") != null ||
```
Comment (vietj) [inline src/main/java/io/vertx/core/impl/VertxImpl.java:90]: why 2 distinct property names ?

Comment (jvican) [reply src/main/java/io/vertx/core/impl/VertxImpl.java:90]: The second property name is a legacy one still used by some applications. I can remove it if you want but I preferred to add it for the sake of completeness, as migrating away from old system properties is not always trivial.

Comment (vietj) [reply src/main/java/io/vertx/core/impl/VertxImpl.java:90]: can you add a comment about it ?

Comment (jvican) [reply src/main/java/io/vertx/core/impl/VertxImpl.java:90]: Sure thing 👍

Comment (jvican) [reply src/main/java/io/vertx/core/impl/VertxImpl.java:91]: Done @vietj

**Comment by vietj** [comment]
can you back-port to 3.9 branch ?

**Comment by vietj** [comment]
thanks for the contribution

**Comment by jvican** [comment]
@vietj sure thing, will send a PR to backport this to 3.9 shortly. Thanks for the help seeing this through.

**Pull request merged**